### PR TITLE
docs: for metasrv maintenance endpoints

### LINKED ISSUE #848: Update docs for feat(metasrv): implement maintenance
A document change request is generated from
https://github.com/GreptimeTeam/greptimedb/pull/3527

### PR DESCRIPTION
**Pull request by tisonkun**
## What's Changed in this PR

This closes #848.

## Checklist

- [x] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [x] This change requires follow-up update in localized docs.

### CODE REVIEW

**Comment by MichaelScofield** [comment]
Lack of Chinese version.

**Comment by tisonkun** [comment]
@MichaelScofield Updated. The previous "Chinese version" written in English so I follow the pattern for now 🤣 

cc @nicecui

**Comment by MichaelScofield** [comment]
But why is the Chinese version also written in English?

**Comment by tisonkun** [comment]
I think it's for historical reason. We should do a follow-up to translate all of admin-api to Chinese. But I tend not to block this PR.

**Comment by tisonkun** [comment]
Or I can translate the new added content anyway.

**Comment by nicecui** [comment]
that's a history problem, pls translate it to Chinese, thank you very much!

**Comment by tisonkun** [comment]
@nicecui I can schedule to do it in this week, but in another PR. Just don't like to couple multiple things in one PR ..

**Review by nicecui — COMMENTED** [review]
Maybe it is a bit late to discuss the APIs in this document PR...However `/maintenance` and `/maintenance/set` are not following the RESTful style of APIs...They should be:

```
# Get the maintenance status
GET /maintenance
# Modify the maintenance status
PUT /maintenance {status:true}
```

Then we can write the document as the following:

```
## /maintenance HTTP endpoint

The metasrv will ignore detected region failures when under maintenance. This is useful when the datanodes are planned to be unavailable for a short period of time; for example, rolling upgrade for datanodes.

### GET

The `/maintenance` endpoint accepts GET HTTP requests and you can use this endpoint to query the maintenance status of your metasrv instance.

curl -X GET http://localhost:3002/admin/maintenance

### POST
.....
```

**Comment by tisonkun** [comment]
@nicecui our `HttpHandler` abstraction doesn't pass the HttpMethod:

```rust
#[async_trait::async_trait]
pub trait HttpHandler: Send + Sync {
    async fn handle(
        &self,
        path: &str,
        params: &HashMap<String, String>,
    ) -> crate::Result<http::Response<String>>;
}
```

**Comment by tisonkun** [comment]
But we can change it anyway.

**Comment by tisonkun** [comment]
DB side changed. Let me update the docs today ..

**Comment by tisonkun** [comment]
Generally we should add this doc to v0.8 :D

**Comment by tisonkun** [comment]
@MichaelScofield @nicecui Updated.

File: docs/v0.7/en/contributor-guide/metasrv/admin-api.md [hunk 1]
```diff
@@ -242,3 +243,51 @@ curl -X GET 'http://localhost:3002/admin/heartbeat?addr=127.0.0.1:4100'
...
+```text
+Maintenance mode is disabled
+```
+
+### PUT
+
+The `/maintenance` endpoint accepts PUT HTTP requests and you can toggle the maintenance status of your metasrv instance.
+
+```bash
+curl -X GET http://localhost:3002/admin/maintenance/set
```
Comment (nicecui) [inline docs/v0.7/en/contributor-guide/metasrv/admin-api.md:276]:
```suggestion
curl -X PUT http://localhost:3002/admin/maintenance
```

File: docs/v0.7/en/contributor-guide/metasrv/admin-api.md [hunk 2]
```diff
@@ -242,3 +243,51 @@ curl -X GET 'http://localhost:3002/admin/heartbeat?addr=127.0.0.1:4100'
...
+```
+
+| Query String Parameter | Type   | Optional/Required | Definition                |
+|:-----------------------|:-------|:------------------|:--------------------------|
+| enable                 | String | Required          | 'true' or 'false'         |
+
+#### Request
+
+```bash
+curl -X GET http://localhost:3002/admin/maintenance/set?enable=true
```
Comment (nicecui) [inline docs/v0.7/en/contributor-guide/metasrv/admin-api.md:286]:
```suggestion
curl -X PUT http://localhost:3002/admin/maintenance?enable=true
```

File: docs/v0.7/zh/contributor-guide/metasrv/admin-api.md [hunk 3]
```diff
@@ -242,3 +243,51 @@ curl -X GET 'http://localhost:3002/admin/heartbeat?addr=127.0.0.1:4100'
...
+```text
+Maintenance mode is disabled
+```
+
+### PUT
+
+The `/maintenance` endpoint accepts PUT HTTP requests and you can toggle the maintenance status of your metasrv instance.
+
+```bash
+curl -X GET http://localhost:3002/admin/maintenance/set
```
Comment (nicecui) [inline docs/v0.7/zh/contributor-guide/metasrv/admin-api.md:276]:
```suggestion
curl -X PUT http://localhost:3002/admin/maintenance
```

File: docs/v0.7/zh/contributor-guide/metasrv/admin-api.md [hunk 4]
```diff
@@ -242,3 +243,51 @@ curl -X GET 'http://localhost:3002/admin/heartbeat?addr=127.0.0.1:4100'
...
+```
+
+| Query String Parameter | Type   | Optional/Required | Definition                |
+|:-----------------------|:-------|:------------------|:--------------------------|
+| enable                 | String | Required          | 'true' or 'false'         |
+
+#### Request
+
+```bash
+curl -X GET http://localhost:3002/admin/maintenance/set?enable=true
```
Comment (nicecui) [inline docs/v0.7/zh/contributor-guide/metasrv/admin-api.md:286]:
```suggestion
curl -X PUT http://localhost:3002/admin/maintenance?enable=true
```

**Comment by tisonkun** [comment]
@nicecui Thanks for the good catches. Updated.

**Review by nicecui — APPROVED** [review]

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**